dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/common/spanexporter/InMemorySpanExporter.java

```diff
@@ -227,8 +227,19 @@ public CompletableResultCode export(Collection<SpanData> spans) {
         Iterator<SpanData> iter = lSpans.listIterator();
         while (iter.hasNext()) {
             SpanData span = iter.next();
-            // Remove readspans and FAT servlet spans
-            if (span.getName().contains("readspans") || span.getName().matches("^/.*/test.*$")) {
+            /*
+             * Remove readspans and FAT servlet spans.
+             *
+             * REGEX breakdown:
+             * Match start of line.
+             * Match 0 or 1 of the string "GET " (note whitespace) using a non-capturing group.
+             * Match a literal "/"
+             * Match anything
+             * Match a literal /
+             * match the string "test"
+             * Match anything until the end of the line.
+             */
+            if (span.getName().contains("readspans") || span.getName().matches("^(?:GET )?/.*/test.*$")) {
                 iter.remove();
             }
```

Comment:
The commit message says you don't know why this was passing before. Have you checked for a build break defect, it may be able to be closed now.

Please also mention this in the PR description.

Reply:
The commit message is out of date (I'll change it), it was the ASL problem I mentioned this morning.

Reply:
ASL = ApplicationStateListener. In the current code, the listener isn't registered properly and doesn't fire.